youtube_mcp.py

```diff
@@ -0,0 +1,296 @@
+"""
+YouTube Download MCP Server using token.pickle authentication.
+
+This MCP server provides tools for downloading YouTube videos owned by 
+the resource owner using token.pickle for authentication.
+"""
+
+import os
+import pickle
+import subprocess
+from pathlib import Path
+from typing import Any
+
+from google.auth.transport.requests import Request
+from google.oauth2.credentials import Credentials
+from googleapiclient.discovery import build
+from mcp.server import FastMCP
+
+# Initialize FastMCP server
+mcp = FastMCP("YouTube Downloader")
+
+def get_authenticated_service(token_pickle_path: str = "token.pickle"):
```

Comment:
We probably need to assume that token.pickle is stored as an environment secret somehow. Not sure how. Maybe there's a textual representation we can use and do some kind of conversion.